src/app/games/siam/SiamMinimax.ts

```diff
@@ -19,17 +19,33 @@ export class SiamMinimax extends Minimax<SiamMove, SiamState, SiamLegalityInform
             for (let x: number = 0; x < 5; x++) {
                 const piece: SiamPiece = node.gameState.getPieceAtXY(x, y);
                 if (piece.belongTo(currentPlayer)) {
-                    moves = moves.concat(SiamRules.getMovesFrom(node.gameState, piece, x, y));
+                    moves = moves.concat(SiamRules.get().getMovesFrom(node.gameState, piece, x, y));
                 }
             }
         }
-        if (node.gameState.countPlayerPawn() < 5) {
-            // up to 20 pushing insertion
-            moves = moves.concat(SiamRules.getPushingInsertions(node.gameState));
-            // up to 24 deraping insertion
-            moves = moves.concat(SiamRules.getDerapingInsertions(node.gameState));
+        if (node.gameState.countCurrentPlayerPawn() < 5) {
+            // up to 44 insertions
+            // we remove some legal but useless insertions as explained below
+            for (const insertion of SiamRules.get().getInsertions(node.gameState)) {
+                if (insertion.direction.get().getOpposite() === insertion.landingOrientation) {
+                    // this is an insertion with an orientation opposite to its direction,
+                    // these are always a useless move and we don't want to take them into account here
+                } else if (this.isInCorner(insertion) && insertion.direction.get() !== insertion.landingOrientation) {
+                    // this insertion is made in the corner but is not forward, so it cannot push
+                    // there is always an equivalent insertion from the other entrance to the same corner,
+                    // but the other one is able to push so it is strictly better
+                } else {
+                    moves.push(insertion);
+                }
+            }
```

Comment:
1. p-e un continue explicite est préférable au if vide ?
2. if insertionsLandingCase == coin : then the two good orientation; else the three ? C'pas plus simple ainsi ?